File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/StatusInput.java
package com.tencent.devops.scm.api.pojo;

import com.tencent.devops.scm.api.enums.StatusState;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StatusInput {
    private StatusState state;
    private String label;
    private String desc;
    private String target;

    /* 这两个字段是工蜂特有 */
    private Boolean block;
    private List<String> targetBranches;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/StatusInput.kt
package com.tencent.devops.scm.api.pojo

import com.tencent.devops.scm.api.enums.StatusState

/**
 * 状态输入信息
 */
data class StatusInput(
    // 状态值
    var state: StatusState,
    // 标签
    var label: String? = null,
    // 描述
    var desc: String,
    // 目标
    var targetUrl: String,
    // 这两个字段是工蜂特有
    // 是否阻塞
    var block: Boolean? = null,
    // 目标分支列表
    var targetBranches: List<String>? = null
)
